File: 206-reverse-linked-list/206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        
        if ( !head || !(head -> next) ) return head;
        
        ListNode* prev = nullptr;
        ListNode* node = head;
        
        while (node) {
            ListNode* saveNext = node -> next;
            node -> next = prev;
            prev = node;
            node = saveNext;
        }
        return prev;
    }
};

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
     
    if (head === null || head.next === null) return head;
    
    let prev = null;
    let node = head;
    
    while (node) {
        let saveNext = node.next;
        node.next = prev
        prev = node
        node = saveNext
    }
    return prev
};
